### PR TITLE
logger: Add 1st arm to shutdown

### PR DESCRIPTION
**Pull request by muramura**
### Solved Problem

None

### Solution

I would like to automatically create a flight logbook by getting the flight time, flight start time, flight end time, flight start location, etc. from ULOG.
I want one ULOG file from first arm to power off.
I live in Japan.
The flight time of the drone in Japan country is assumed to be from "first takeoff to power off".
The drone in Japan country may move without turning off the power in the state of arm release.
PX4-Autopilot logs from boot to power off.
I may only power on and not arm.
I can create a valid ULOG file.

### Alternatives

None

### Test coverage

None

### Context

Logger menu
Screenshot from 2023-01-15 19-17-37

Command start
Screenshot from 2023-01-15 19-27-09

Description
Screenshot from 2023-01-15 19-27-23

Parameter selection
Screenshot from 2023-01-15 19-28-39
Screenshot from 2023-01-15 19-31-22

ULOG: start Armed!! and meny armed, disarmed..
Screenshot from 2023-01-15 19-36-31

### CODE REVIEW

**Review by AlexKlimaj — APPROVED** [review]

File: src/modules/logger/logger.cpp [hunk 1]
```diff
@@ -2405,6 +2413,7 @@ Or if already running:
 	PRINT_MODULE_USAGE_COMMAND("start");
 	PRINT_MODULE_USAGE_PARAM_STRING('m', "all", "file|mavlink|all", "Backend mode", true);
 	PRINT_MODULE_USAGE_PARAM_FLAG('x', "Enable/disable logging via Aux1 RC channel", true);
+	PRINT_MODULE_USAGE_PARAM_FLAG('a', "Log 1st armed until shutdown (implies -e)", true);
```
Comment (bkueng) [inline src/modules/logger/logger.cpp:2416]:
```suggestion
	PRINT_MODULE_USAGE_PARAM_FLAG('a', "Log 1st armed until shutdown", true);
```

File: src/modules/logger/logger.cpp [hunk 2]
```diff
@@ -1122,6 +1126,10 @@ bool Logger::start_stop_logging()
 
 	// only start/stop if this is a state transition
 	if (updated && _prev_state != desired_state) {
+		if (_prev_state && _log_mode == LogMode::arm_until_shutdown) {
+			return true;
+		}
```
Comment (bkueng) [inline src/modules/logger/logger.cpp:1131]: Can you move this logic to https://github.com/PX4/PX4-Autopilot/blob/5bff4f19e44a455968db381bf33d78a75223cfb8/src/modules/logger/logger.cpp#L1120 and update `desired_state` instead:
`desired_state = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) || (_prev_state && _log_mode == LogMode::arm_until_shutdown);`

Comment (muramura) [reply src/modules/logger/logger.cpp:1131]: I agree.

**Review by bkueng — APPROVED** [review]
Thanks

**Pull request merged**